### PR TITLE
Add default movies from company in search page

### LINKED ISSUE #120: Simplify film search
We want to make search screen easier to use with less input. We do this by:
- simplifying search form (less input)
- providing meaningful defaults (movies produced by current RH)

## Scope
- replace current movie search form with a single search box capable of searching by Movie title and by IMDB (the field should determine that intelligently on it is own)
- change the screen layout so that the search field is on top of the result table
- show all the movies for the current RH/Licensee by default. Use the same `discover` endpoint as described in #93. This feature supersedes #79 
- show IMDB number for each of the search results

## Roles
gardener: @troggy 100 DAI
worker: @mwaa 370 DAI
reviewer: @troggy 80 DAI

### PR DESCRIPTION
**Pull request by mwaa**
Please remember the [Contributing Guidelines](https://WhiteRabbitTheWorldIsYours/dashboard/blob/master/CONTRIBUTION.md) :heart:

## What does this PR do?

Simplifies movie search

## Recommendations for testing

- Use Production company rightsholder account
- Click on add movies
- The Search page should list movies that belong to the rights holder company
- There should be a single search field on the top right of screen. You can use text or imdb id to search for movies
- Movies listed after a search

## Links to relevant issues or information

Closes #120

### CODE REVIEW

File: frontend/src/stores/API.ts [hunk 1]
```diff
@@ -171,9 +172,9 @@ export const searchMovies = async (id: string, title: string, year: string)
     }
   }
 
-  if (title !== 'undefined' && title !== '') {
+  if (query !== 'undefined' && query !== '') {
```
Comment (troggy) [inline frontend/src/stores/API.ts:175]: if `query` is empty we don't want to execute this function at all. I suggest to shortcut the function with `if (!query) return [];` guard at the beginning plus avoid calling it whatsoever for empty queries (see the other comment)

File: frontend/src/containers/MovieSearch.tsx [hunk 2]
```diff
@@ -44,9 +48,24 @@ const columns = [
...
 export default () => {
   let loading = false;
+  const { user } = useContext(DashboardContext);
   const [movie, setMovie] = useState<MovieInterface>();
   const [display, setDisplay] = useState<MovieInterface[]>();
 
+  useEffect(() => {
+    if (user) {
+      getCompanyMovies(user.company.id)
+        .then((tmdbMovies) => setDisplay(tmdbMovies.map((current: any) => ({
```
Comment (troggy) [inline frontend/src/containers/MovieSearch.tsx:58]: consider using the common [tmdbToMovie](https://github.com/WhiteRabbitTheWorldIsYours/dashboard/blob/master/common/src/models.ts#L165) helper here. [Example](https://github.com/WhiteRabbitTheWorldIsYours/dashboard/blob/master/frontend/src/containers/components/PendingUserScreen.tsx#L18)

File: frontend/src/stores/API.ts [hunk 3]
```diff
@@ -171,9 +172,9 @@ export const searchMovies = async (id: string, title: string, year: string)
...
     const { results } = await fetch(
```
Comment (troggy) [inline frontend/src/stores/API.ts:176]: this is being called for `tt...` queries as well, consder `return`ing value right away after the `getMovieDetails` call above

**Review by troggy — CHANGES_REQUESTED** [review]
tested the functionality: works great 👍

Left some comments on the style and efficiency of the code

File: frontend/src/containers/MovieSearch.tsx (hunk 2 above)
Comment (mwaa) [reply frontend/src/containers/MovieSearch.tsx:59]: Can't use this now made comment for standardize movie interfaces

**Review by troggy — APPROVED** [review]
👍

**Pull request merged**